src/dsync/dsync.c

```diff
@@ -2411,17 +2411,21 @@ static void dsync_disjunction_free(struct dsync_disjunction* disjunction)
 {
     struct dsync_conjunction *conjunction;
     struct dsync_conjunction *n;
-
-    assert(list_empty(&disjunction->linkage));
-    list_for_each_entry_safe(conjunction,
-                             n,
-                             &disjunction->conjunctions,
-                             linkage) {
-        list_del_init(&conjunction->linkage);
-        dsync_conjunction_free(conjunction);
+    
+    if (disjunction) {
```

Comment:
style: This could be simplified a bit with an early exit:

```suggestion
    if (!disjunction) {
        return;
        }
```

Then the code after that (`assert()`, `list_for_each_entry_safe()`, etc.) doesn't change at all.

Reply:
I did that first and took it out. I'll put it back